playwright/src/main/java/com/microsoft/playwright/impl/LocatorImpl.java

```diff
@@ -19,7 +19,7 @@
 import static com.microsoft.playwright.impl.Utils.convertType;
 import static com.microsoft.playwright.impl.Utils.toJsRegexFlags;
 
-class LocatorImpl implements Locator {
+public class LocatorImpl implements Locator {
```

Comment:
This class is an implementation detail and is not intended to be extended by third parties. Can you start with describing what are you trying to achieve?

Reply:
I am adding support of playwright to [qaf](https://qmetry.github.io/qaf) - well accepted framework for [selenium](https://www.selenium.dev/ecosystem/#frameworks). It's required to extend original implementation, In order to support features like [custom component](https://qmetry.github.io/qaf/latest/custom_component.html), [locator repository](https://qmetry.github.io/qaf/latest/locator_repository.html), self-descriptive locator.

Reply:
It's not clear from the docs you linked why the features described there would "require to extend" the implementation rather than just using it as a client. We expect all clients to only depend on the public API and use aggregation rather than inheriting from the implementation. Even if you extended LocatorImpl, it would not be interoperable with the rest of the framework (e.g. Page.locator would still return builtin locator). I suggest you start with an issue describing the problem and why it cannot be addressed with current API.